### PR TITLE
DAOS-8012 tests: allow test_daos_server_dump_basic to run in CI

### PR DESCRIPTION
**Pull request by bfaccini**
Now that test_daos_server_dump_basic intermittent/timing
issue has been fixed by PR-7768, allow it to be run by
CI.

Change-Id: Ic0cbc7f7ffca1c6280802cb342773185f474f883
Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

### CODE REVIEW

**Review by daltonbohning — CHANGES_REQUESTED** [review]
The test needs to be ran before merging, since it is not in the default set of `pr` tests
```
Test-tag: test_daos_server_dump_basic
```

**Comment by bfaccini** [comment]
> The test needs to be ran before merging, since it is not in the default set of `pr` tests
> 
> ```
> Test-tag: test_daos_server_dump_basic
> ```

done

**Review by daltonbohning — CHANGES_REQUESTED** [review]
`Test-tag: pr,test_daos_server_dump_basic` specifies an intersection of `pr` and `test_daos_server_dump_basic`. But `test_daos_server_dump_basic` is in `daily_regression`, not `pr`. You should use either
- `Test-tag: test_daos_server_dump_basic` - Runs just this test
- `Test-tag: pr test_daos_server_dump_basic` - Runs all `pr` tests and this test

**Comment by daltonbohning** [comment]
> Ah,... Is it a recent behaviour of the Test-tag ?? Thx for pointing it to me and for your details !! ;-) This is soooooo unclear OTOH :-(

This is what the behavior has always been. The syntax is what Avocado uses. This page explains it: [Test-Tags](https://daosio.atlassian.net/wiki/spaces/DAOS/pages/10984259629/Test+Tags#How-Does-Avocado-Handle-Sets-of-Tags?)
Please feel free to update/comment on that page is something still isn't clear

In short:
- `pr,my_test` runs tests tagged with BOTH `pr` and `my_test`
- `pr my_test` runs tests tagged with `pr` OR `my_test`

**Review by daltonbohning — DISMISSED** [review]

**Comment by bfaccini** [comment]
@sylviachanoiyee and @phender any chance that you can review ?

**Comment by bfaccini** [comment]
"Functional Hardware Medium/ REBUILD4" single/only failure in last CI run is unrelated to this PR and is definitely being addressed in [DAOS-10832](https://daosio.atlassian.net/browse/DAOS-10832)

**Comment by bfaccini** [comment]
@sylviachanoiyee and @phender , any chance you can review this ??

File: src/tests/ftest/server/daos_server_dump.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ def test_daos_server_dump_basic(self):
         Test Description: Test engine ULT stacks dump (basic).
 
         :avocado: tags=all
+        :avocado: tags=vm,daily_regression
```
Comment (phender) [inline src/tests/ftest/server/daos_server_dump.py:57]: Minor, but our convention is to include when to run the functional test in the first tag line.
```suggestion
        :avocado: tags=all,daily_regression
        :avocado: tags=vm
```

**Review by phender — CHANGES_REQUESTED** [review]

**Comment by phender** [comment]
To get a quicker turnaround in CI, you can use these commit pragmas:
```
Skip-unit-tests: true
Test-tag: test_daos_server_dump_basic
```

**Review by sylviachanoiyee — DISMISSED** [review]

File: src/tests/ftest/server/daos_server_dump.py (hunk 1 above)
Comment (bfaccini) [reply src/tests/ftest/server/daos_server_dump.py:57]: ok, but with "when to run the functional test" you mean separate the set of "timing" tags from the "config" tags, and put them first, right ?

**Comment by bfaccini** [comment]
> To get a quicker turnaround in CI, you can use these commit pragmas:
> 
> ```
> Skip-unit-tests: true
> Test-tag: test_daos_server_dump_basic
> ```

Ok, thx for the tip, will do

**Review by sylviachanoiyee — APPROVED** [review]

**Comment by bfaccini** [comment]
Oops, and now the all the builds in CI are failing due to spdk_pci_get_first_device() declaration missing :-( will try to again merge with latest master ...

**Review by phender — APPROVED** [review]

**Comment by phender** [comment]
@bfaccini the https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-9219/6/artifact/Functional%20on%20EL%208/ does not include any results for the server/daos_server_dump.py test.  You're most recent commit was missing this pragma:
```
Test-tag: pr test_daos_server_dump_basic
```

File: src/tests/ftest/server/daos_server_dump.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/server/daos_server_dump.py]: Right, put another way the format we follow is:
```
:avocado: tags=<when_to_run>      # e.g. 'all' + 'pr', 'daily_regression', or 'full_regression' (weekly)
:avocado: tags=<where_to_run>     # e.g. test stage used to run this test, e.g. 'vm', 'hw,small', 'hw,medium', or 'hw,large'
:avocado: tags=<feature_tags>     # tags used with the 'Features: <tag>' pragma
:avocado: tags=<test_tags>        # one or more tags used to select this test. Must include one unique tag (usually the full test method name)
```

**Pull request merged**